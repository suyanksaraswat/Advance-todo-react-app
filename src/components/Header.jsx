import React from 'react';
import logo from '../logo.svg';
import './../App.css';
import './../fonts/font-awesome.css';

type HeaderProps = {
	length: Number,
};

const Header = (props: HeaderProps) => {
	return (
		<>
			<nav className="navbar fixed-top navbar-dark bg-dark">
				<img src={logo} className="App-logo" alt="logo" />
				<h1 className="navbar-brand">
					Todo Count: <span className="badge badge-pill badge-primary">{props.length}</span>
				</h1>
			</nav>
		</>
	)
}

export default Header
