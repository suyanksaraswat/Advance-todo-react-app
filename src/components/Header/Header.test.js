import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashing", () => {
	const div = document.createElement("div");
	ReactDOM.render(<Header></Header>, div)
})

it("renders TodoInput correctly 1", () => {
	const {getByTestId} = render(<Header length='2'></Header>)
	expect(getByTestId('header')).toHaveTextContent("Todo Count: 2")
})

it("renders TodoInput correctly 2", () => {
	const {getByTestId} = render(<Header length='10'></Header>)
	expect(getByTestId('header')).toHaveTextContent("Todo Count: 10")
})

it("matches snapshot 1", () => {
	const tree = renderer.create(<Header length='10'></Header>).toJSON();
	expect(tree).toMatchSnapshot();
})

it("matches snapshot 2", () => {
	const tree = renderer.create(<Header length='2'></Header>).toJSON();
	expect(tree).toMatchSnapshot();
})
