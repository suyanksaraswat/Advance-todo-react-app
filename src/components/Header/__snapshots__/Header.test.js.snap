// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  data-testid="header"
>
  <nav
    className="navbar fixed-top navbar-dark bg-dark"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="navbar-brand"
    >
      Todo Count: 
      <span
        className="badge badge-pill badge-primary"
      >
        10
      </span>
    </h1>
  </nav>
</div>
`;

exports[`matches snapshot 2 1`] = `
<div
  data-testid="header"
>
  <nav
    className="navbar fixed-top navbar-dark bg-dark"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="navbar-brand"
    >
      Todo Count: 
      <span
        className="badge badge-pill badge-primary"
      >
        2
      </span>
    </h1>
  </nav>
</div>
`;
